EIPS/eip-4762.md

```diff
@@ -62,10 +70,10 @@ If a `*CALL` or `SELFDESTRUCT` is value-bearing (ie. it transfers nonzero wei),
 
 Note: when checking for the existence of the `callee`, the existence check is done by validating that there is an extension-and-suffix tree at the corresponding stem, and does not rely on `CODEHASH_LEAF_KEY`.
 
-When calling `EXTCODEHASH`, process the access event:
+When calling `*CALL`, `SELFDESTRUCT`, `EXTCODESIZE`, `EXTCODECOPY` or `EXTCODEHASH` on an EIP-7702 delegating contract, process the additional access event:
 
 ```
-(address, 0, CODEHASH_LEAF_KEY)
+(delegated_address, 0, CODEHASH_LEAF_KEY)
```

Comment:
I'm a bit confused by this change.
For example, if you're executing an `EXTCODESIZE` as mentioned above:
- We should provide the first-code chunk to allow a stateless client to detect if there's a delegation and setup the right context of the target.
- If there is a delegation, we'll include the `BASIC_DATA` of the delegated address which contains the code size.

Why there is a need for this `CODEHASH_LEAF_KEY` of the delegated address?

Reply:
More generally, I think we should explain impact on the witness for the delegation resolving step. Then that might change the context of which address is used for the EXTCODEHASH/EXTCODESIZE/etc and simply apply already existing rules to include the witness for that address (which now can be `address` or `delegated_address`)